### PR TITLE
Replacing ShoppingListCell xib

### PR DESCRIPTION
**Pull request by mobileapps23**

### CODE REVIEW

File: Snabble/UI/Scanner/ShoppingListCell.swift [hunk 1]
```diff
@@ -107,17 +90,17 @@ extension ShoppingListCell {
...
-        self.spinner.isHidden = false
+        self.cellView?.spinnerStartAnimation()
+        self.cellView?.spinnerIsHidden(false)
         self.task = Snabble.urlSession.dataTask(with: url) { data, _, error in
             self.task = nil
             DispatchQueue.main.async {
-                self.spinner.stopAnimating()
-                self.spinner.isHidden = true
+                self.cellView?.spinnerStopAnimation()
+                self.cellView?.spinnerIsHidden(true)
```
Comment (mrgrauel) [inline Snabble/UI/Scanner/ShoppingListCell.swift:103]: I would recommend to use [`hidesWhenStopped`](https://developer.apple.com/documentation/uikit/uiactivityindicatorview/1622837-hideswhenstopped#) instead of `spinnerIsHidden(_:)`.

File: Snabble/UI/Scanner/ShoppingListCellView.swift [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            quantityLabel.leadingAnchor.constraint(equalTo: tickLayoutGuide.leadingAnchor),
+            quantityLabel.trailingAnchor.constraint(equalTo: checkContainer.leadingAnchor),
+            quantityLabel.heightAnchor.constraint(equalToConstant: 24),
+            quantityLabel.centerYAnchor.constraint(equalTo: tickLayoutGuide.centerYAnchor)
+        ])
+    }
+
+    public func prepareForReuse() {
+        self.productImage?.image = nil
+        self.spinner?.isHidden = true
```
Comment (mrgrauel) [inline Snabble/UI/Scanner/ShoppingListCellView.swift:144]: This can also be removed if we use `hidesWhenStopped`.

File: Snabble/UI/Scanner/ShoppingListCellView.swift [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        self.productImage?.image = image
+    }
+
+    public func spinnerStartAnimation() {
+        self.spinner?.startAnimating()
+    }
+
+    public func spinnerStopAnimation() {
+        self.spinner?.stopAnimating()
+    }
```
Comment (mrgrauel) [inline Snabble/UI/Scanner/ShoppingListCellView.swift:170]: If the `spinner` is not private we don't need these methods.
If you want to make the `spinner` private it's fine to use these methods, but I would rename them to `startAnimating()` and `stopAnimating()` to match the activityIndicator interface.

File: Snabble/UI/Scanner/ShoppingListCellView.swift [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    public func spinnerIsHidden(_ bool: Bool) {
+        self.spinner?.isHidden = bool
+    }
```
Comment (mrgrauel) [inline Snabble/UI/Scanner/ShoppingListCellView.swift:174]: Is hot needed if you use `hidesWhenStopped`.

File: Snabble/UI/Scanner/ShoppingListCellView.swift [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        super.init(frame: frame)
+        self.setupUI()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupUI() {
+        self.backgroundColor = .clear
```
Comment (mrgrauel) [inline Snabble/UI/Scanner/ShoppingListCellView.swift:30]: Is that really needed? I don't see a reason to set a `backgroundColor`.

File: Snabble/UI/Scanner/ShoppingListCell.swift [hunk 6]
```diff
@@ -10,57 +10,46 @@ final class ShoppingListCell: UITableViewCell {
...
-        checkContainer.layer.shadowOpacity = 0.3
-        checkContainer.layer.shadowOffset = CGSize.zero
-        checkContainer.layer.shadowRadius = 1.5
-        checkContainer.layer.masksToBounds = true
-        checkContainer.layer.borderWidth = 1 / UIScreen.main.scale
-        checkContainer.layer.borderColor = UIColor.systemBackground.cgColor
-        checkContainer.layer.cornerRadius = checkContainer.bounds.width / 2
-        checkContainer.backgroundColor = .clear
+    private func setupUI() {
+        self.backgroundColor = .clear
```
Comment (mrgrauel) [inline Snabble/UI/Scanner/ShoppingListCell.swift:29]: Is that really needed? I don't see a reason to set a `backgroundColor`.

**Review by mrgrauel — CHANGES_REQUESTED** [review]

File: Snabble/UI/Scanner/ShoppingListCellView.swift [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+        let spinner = UIActivityIndicatorView()
+        spinner.translatesAutoresizingMaskIntoConstraints = false
+        spinner.hidesWhenStopped = true
+        if #available(iOS 13.0, *) {
+            spinner.style = .medium
+        } else {
+            spinner.style = .gray
+        }
+        spinner.contentMode = .scaleToFill
```
Comment (mrgrauel) [inline Snabble/UI/Scanner/ShoppingListCellView.swift:45]: Is this needed? I don't see a reason for it. Why did you add it?

File: Snabble/UI/Scanner/ShoppingListCellView.swift [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+            imageViewLayuotGuide.bottomAnchor.constraint(equalTo: bottomAnchor),
+
+            productImage.leadingAnchor.constraint(equalTo: imageViewLayuotGuide.leadingAnchor, constant: 14),
+            productImage.widthAnchor.constraint(equalToConstant: 28),
+            productImage.heightAnchor.constraint(equalToConstant: 28),
+            productImage.centerYAnchor.constraint(equalTo: imageViewLayuotGuide.centerYAnchor),
+
+            spinner.leadingAnchor.constraint(equalTo: imageViewLayuotGuide.leadingAnchor),
+            spinner.widthAnchor.constraint(equalToConstant: 28),
+            spinner.heightAnchor.constraint(equalToConstant: 28),
```
Comment (mrgrauel) [inline Snabble/UI/Scanner/ShoppingListCellView.swift:111]: The Activity Indicator should be centered in the box. It's pinned to the left at the moment. Please also add contraints to each boundary if you center it.

**Review by mrgrauel — CHANGES_REQUESTED** [review]

File: Snabble/UI/Scanner/ShoppingListCellView.swift (hunk 7 above)
Comment (mobileapps23) [reply Snabble/UI/Scanner/ShoppingListCellView.swift:45]: Adding `spinner.style = .medium` gives an error _"medium' is only available in iOS 13.0 or newer"_, as SDK has current deployment target iOS version 12.4. Otherwise, using only `spinner.style = .gray` is deprecated for iOS 13.0+ and it's recommended to use `.medium` instead. So while SDK iOS version is lower than 13.0, we should define both cases of styling.

Comment (mrgrauel) [reply Snabble/UI/Scanner/ShoppingListCellView.swift:45]: I just mean line 45 `spinner.contentMode = .scaleToFill`

File: Snabble/UI/Scanner/ShoppingListCellView.swift (hunk 8 above)
Comment (mobileapps23) [reply Snabble/UI/Scanner/ShoppingListCellView.swift:108]: Fixed: `spinner` centered by both axis related to `productImage`

File: Snabble/UI/Scanner/ShoppingListCellView.swift (hunk 7 above)
Comment (mobileapps23) [reply Snabble/UI/Scanner/ShoppingListCellView.swift]: I'm sorry for the misunderstanding, removed content mode case, as it's not needed.

File: Snabble/UI/Scanner/ShoppingListCellView.swift (hunk 8 above)
Comment (mrgrauel) [reply Snabble/UI/Scanner/ShoppingListCellView.swift:109]: 1. That's good, but could you add constraints to all four edges, that is never if an image is too small is drawn out of bounds?
2. Do we need the height and width constraint for the spinner? I don't think so cause it' has an `intrinsicContentSize`.

File: Snabble/UI/Scanner/ShoppingListCell.swift (hunk 6 above)
Comment (mrgrauel) [reply Snabble/UI/Scanner/ShoppingListCell.swift]: I removed it in a3936c529bf96c23730e2e40de0e9a2a3bcfc23e
